src/core_init_atmosphere/mpas_stack.F

```diff
@@ -0,0 +1,273 @@
+   !> polymorphic data structure to provide usage in different areas.
+   !>
+   !>
+   !> Creating a Stack
+   !> ==================
+   !> The stack data structure (`type (node)`) is defined by a single `next` pointer
+   !> and a pointer to a `type (payload_t)`, which is defined as a empty derived type.
+   !>
+   !> To use the stack, create a derived that extends the `payload_t` type.  Define your
+   !> extended derived type with members that meets your application.
+   !>
+   !> For instance:
+   !> ```
+   !> type, extends(payload_t) :: my_payload_name
+   !>    ! Define the members of your type as you wish
+   !> end type my_payload_name
+   !>
+   !> class (my_payload_name) :: item1, item2
+   !> ```
+   !>
+   !> The extended payload_t define type will enable it to ride along with the stack.
+   !> It will also enable you to push the same payload twice (if need-be).
+   !>
+   !> You will then need to create a stack (or multiple stacks if you desire) as
+   !> the following:
+   !>
+   !> ```
+   !> type (node) :: stack1, stack2
+   !> ```
+   !> 
+   !>  Pushing onto a Stack
+   !>  ====================
+   !>  You can push your items onto a stack as:
+   !> 
+   !> ```
+   !> stack1 => mpas_satck_push(stack1, item1)
+   !> ```
+   !> 
+   !> Popping an item off of the stack
+   !> ================================
+   !> Popping an item off of the stack will require a bit more work then pushing.
+   !> Because we've done some fancy Fortran class polymorphism, we will need to 
+   !> use the select case to get our type (or multiple types) back into a usable
+   !> object:
+   !> ```
+   !> ! The item to pop items into
+   !> class (payload_t), pointer :: top
+   !> type (my_payload_name), pointer :: my_item
+   !> 
+   !> top => mpas_stack_pop(stack1)
+   !> select type(top)
+   !>    type is(my_payload_name)
+   !>       my_item => top
+   !> end  select
+   !> ```
+   !> 
+   !> Note: It is recommended to create your own `pop` function so you can limit
+   !> reduce the amount of coded needed. An example is provided at the bottom of
+   !> this module as the function `my_pop(..)`
+   !
+   !-----------------------------------------------------------------------
+
+   contains
+
+   !***********************************************************************
+   !
+   !  routine mpas_stack_is_empty
+   !
+   !> \brief   Returns .TRUE. if the stack is empty, otherwise .FALSE.
+   !> \author  Miles A. Curry 
+   !> \date    04/04/19
+   !> Returns .true. If the stack is empty and/or if the stack is unassociated.
+   !
+   !-----------------------------------------------------------------------
+   function mpas_stack_is_empty(stack) result(is_empty)
+
+      implicit none
+      type (node), intent(in), pointer :: stack
+      logical :: is_empty
+
+      is_empty = .TRUE.
+      if (associated(stack)) then
+         is_empty = .FALSE.
+         return
+      endif
+
+      return
```

Comment:
The two `return` statements in this function could be omitted for simplicity.
Stylistically, the rest of MPAS tends to not use upper-case letters for `.TRUE.` and `.FALSE.`.

Reply:
Sounds great I'll remove the `return` and change the booleans to be lower-case.